hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/protocol/ClientProtocol.java

```diff
@@ -697,6 +697,12 @@ boolean mkdirs(String src, FsPermission masked, boolean createParent)
   DirectoryListing getListing(String src, byte[] startAfter,
       boolean needLocation) throws IOException;
 
+  @Idempotent
+  BatchedDirectoryListing getBatchedListing(
```

Comment:
I believe we want to add the annotation
`  @ReadOnly(isCoordinated = true)
`

Reply:
Good catch. Done.